cmake/modules/Libdispatch.cmake

```diff
@@ -75,6 +75,12 @@ foreach(sdk ${DISPATCH_SDKS})
 
     if(sdk MATCHES WINDOWS)
       set(SWIFT_LIBDISPATCH_COMPILER_TRIPLE_CMAKE_ARGS -DCMAKE_C_COMPILER_TARGET=${SWIFT_SDK_WINDOWS_ARCH_${arch}_TRIPLE};-DCMAKE_CXX_COMPILER_TARGET=${SWIFT_SDK_WINDOWS_ARCH_${arch}_TRIPLE})
+    elseif(sdk MATCHES LINUX)
+      set(SWIFT_LIBDISPATCH_COMPILER_CMAKE_ARGS ${SWIFT_LIBDISPATCH_COMPILER_CMAKE_ARGS};-DCMAKE_SYSROOT=${SWIFT_SDK_${sdk}_ARCH_${arch}_PATH})
+      set(SWIFT_LIBDISPATCH_COMPILER_TRIPLE_CMAKE_ARGS -DCMAKE_C_COMPILER_TARGET=${SWIFT_SDK_${sdk}_ARCH_${arch}_TRIPLE};-DCMAKE_CXX_COMPILER_TARGET=${SWIFT_SDK_${sdk}_ARCH_${arch}_TRIPLE})
+      # Use linker used to build Swift
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -w -fuse-ld=${SWIFT_USE_LINKER}")
```

Comment:
This `Libdispatch.cmake` is loaded by `include(Libdispatch)`, so we should not overwrite `CMAKE_C_FLAGS` here. Could you define a new local scope variable instead?